executor/aggfuncs/func_group_concat.go

```diff
@@ -491,32 +517,49 @@ func (e *groupConcatDistinctOrder) UpdatePartialResult(sctx sessionctx.Context,
 			continue
 		}
 		p.valSet.Insert(joinedVal)
+		memDelta += int64(len(joinedVal))
 		sortRow := sortRow{
 			buffer:  buffer,
 			byItems: make([]*types.Datum, 0, len(e.byItems)),
 		}
 		for _, byItem := range e.byItems {
 			d, err := byItem.Expr.Eval(row)
 			if err != nil {
-				return 0, err
+				return memDelta, err
 			}
+			memDelta += GetDatumMemSize(&d)
 			sortRow.byItems = append(sortRow.byItems, d.Clone())
 		}
+		oldMem := p.topN.currSize
 		truncated := p.topN.tryToAdd(sortRow)
+		newMem := p.topN.currSize
+		memDelta += int64(newMem - oldMem)
 		if p.topN.err != nil {
-			return 0, p.topN.err
+			return memDelta, p.topN.err
 		}
 		if truncated {
 			if err := e.handleTruncateError(sctx); err != nil {
-				return 0, err
+				return memDelta, err
 			}
 		}
 	}
-	return 0, nil
+	return memDelta, nil
 }
 
 func (e *groupConcatDistinctOrder) MergePartialResult(sctx sessionctx.Context, src, dst PartialResult) (memDelta int64, err error) {
 	// If order by exists, the parallel hash aggregation is forbidden in executorBuilder.buildHashAgg.
 	// So MergePartialResult will not be called.
 	return 0, terror.ClassOptimizer.New(mysql.ErrInternal, mysql.MySQLErrName[mysql.ErrInternal]).GenWithStack("groupConcatDistinctOrder.MergePartialResult should not be called")
 }
+
+func GetDatumMemSize(d *types.Datum) int64 {
```

Comment:
Is `GetDatumMemSize` not need to export, and rename it to `getDatumMemSize`?

Reply:
I just want to reuse `GetDatumMemSize` in func_group_concat_test.go,  so expose it. Rename it to `getDatumMemSize` and write same method in func_group_concat_test.go will be better?

Reply:
> Is `GetDatumMemSize` not need to export? Could we rename it to `getDatumMemSize`?

I just want to reuse `GetDatumMemSize` in func_group_concat_test.go, so expose it. Rename it to `getDatumMemSize` and write same method in func_group_concat_test.go will be better?

Reply:
ok, you need to add a comment for it to resolve following error.
```
[2020-09-29T08:38:31.415Z] linting
[2020-09-29T08:38:31.981Z]   ✘  https://revive.run/r#exported  exported function GetDatumMemSize should have comment or be unexported  
[2020-09-29T08:38:31.981Z]   executor/aggfuncs/func_group_concat.go:555:1
```